yaas/config.py

```diff
@@ -3,28 +3,39 @@
 from __future__ import absolute_import
 from __future__ import print_function
 
+import pprint
+import requests
+
 # These default configs are overriden based on
-# command line arguments or environment variables
+# command line arguments or environment variables.
 
-server_url = None   # Must be overriden
+scheme = 'http'
+server = None   # Must be overriden
+port = 8080
 username = 'admin'
 password = 'admin'
-args = None
+command = None
+
 
 def print_request_and_response(res, *args, **kwargs):
     print("Request:", res.request.__dict__)
     print("Response:", res.__dict__)
 
-def requests_opts():
-    opts = {
-        'auth': (username, password),
-        'headers': {'X-Requested-By': 'Ambari'},
-        }
-
-    if args.verbose:
-        opts['hooks'] = {
-            'response': print_request_and_response,
-            }
-
-    return opts
 
+def request(endpoint, callback):
```

Comment:
"callback" makes it sound like this is an asynchronous request. Maybe print_function?